### DISCUSSION TITLE
How can I get AnalyzerConfigOptions inside RegisterPostInitializationOutput in an IIncrementalGenerator?

### DISCUSSION THREAD

**Discussion by workgroupengineering** [dotnet/roslyn#71521] [Q&A]
I need get RootNamespace inside RegisterPostInitializationOutput. How do?

**Comment by CyrusNajmabadi** [comment]
Root namespace only affects the ide in terms of what it does in some templates. Can you clarify what you would just it for?

(Note: I don't believe this value is in the editorconfig data either)

**Reply by workgroupengineering** [reply]
I need to create a generator similar to [this](https://gist.github.com/SaahilClaypool/b1aca690f154ec0fe9ed49751988e701). Which instead of having the namespace hardcoded, automatically retrieves the namespace from the project where the generator is referenced.

for accessing `RootNamespace` you need to add the following code in csproj

```xml
  <ItemGroup>
    <CompilerVisibleProperty Include="RootNamespace" />
  </ItemGroup>
```

and in your analyzer/genererator

```c#
 context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.RootNamespace", out var rootNamespace);
```

**Reply by jasonmalinowski** [reply]
RegisterPostInitializationOutput shouldn't depend on the settings like this; this would have to go to regular output.
